src/lib/verify.js

```diff
@@ -52,5 +63,47 @@ export const verify = async (token) => {
     throw new Error("Token expired");
   }
 
-  return { address: address.to_bech32(), body: parsed_body };
+  return {
+    address: address.to_bech32(),
+    network: address.network_id(),
+    body: parsed_body,
+  };
+};
+
+const verifyAddress = (checkAddress, publicKey) => {
+  console.log("publicKey", publicKey.hash());
+  // if (this.headers.address.to_bech32() !== checkAddress.to_bech32()) {
+  //   console.log("FASLE1");
+  //   return false;
+  // }
+  // check if BaseAddress
+  try {
+    const baseAddress = Loader.Cardano.BaseAddress.from_address(checkAddress);
+    //reconstruct address
+    const paymentKeyHash = publicKey.hash();
+    const stakeKeyHash = baseAddress.stake_cred().to_keyhash();
+    const reconstructedAddress = Loader.Cardano.BaseAddress.new(
+      checkAddress.network_id(),
+      Loader.Cardano.StakeCredential.from_keyhash(paymentKeyHash),
```

Comment:
Maybe we could return some kinda "verification object" here, for example:

```javascript
{ 
  verified: false,
  error: "Error message here"
}
```
Then we could throw error from main `verify` function or return failed result with appropriate error message, for example `Address verification failed`  or `Stake address verification failed` etc.

Reply:
Nice thinking; I will update now.